### PR TITLE
Jinan/update use props for doc

### PR DESCRIPTION
**Pull request by PorterNan**
# What

update use props for doc

# Why

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/storybook/stories/Stateful Client/UsePropsForHook.stories.mdx [hunk 1]
```diff
@@ -8,49 +8,76 @@ import { Canvas, Description, Heading, Props, Source, Title } from '@storybook/a
...
+
+- [VideoGallery](/?path=/docs/ui-components-videogallery--video-gallery)
+- [ParticipantList](/?path=/docs/ui-components-participantlist--participant-list)
+- [CameraButton](/?path=/docs/ui-components-controlbar-buttons-camera--camera)
+- [EndCallButton](/?path=/docs/ui-components-controlbar-buttons-endcall--end-call)
+- [MicrophoneButton](/?path=/docs/ui-components-controlbar-buttons-microphone--microphone)
+- [OptionsButton](/?path=/docs/ui-components-controlbar-buttons-options--options)
+- [ScreenShareButton](/?path=/docs/ui-components-controlbar-buttons-screenshare--screen-share)
+
+## Choose your flavors
```
Comment (prprabhu-ms) [inline packages/storybook/stories/Stateful Client/UsePropsForHook.stories.mdx:57]: nit: s/flavors/flavor

Comment (prprabhu-ms) [reply packages/storybook/stories/Stateful Client/UsePropsForHook.stories.mdx:57]: flavor isn't indicative of what the argument does.

Should we call this `target` (i.e., targeting chat vs calling), or `source` (i.e., the source of the bindings).
I like `target` more. The packaging detail is not obvious to the users.

File: packages/storybook/stories/Stateful Client/UsePropsForHook.stories.mdx [hunk 2]
```diff
@@ -8,49 +8,76 @@ import { Canvas, Description, Heading, Props, Source, Title } from '@storybook/a
...
+
+usePropsFor() is context-awared so no need to worry which Provider you are using, there is only one exception:
```
Comment (prprabhu-ms) [inline packages/storybook/stories/Stateful Client/UsePropsForHook.stories.mdx:59]: I'd make this prose a teeny bit more formal.
Something like:

`usePropsFor()` is context-aware. It picks out props from the relevant Provider for the component. One exception is `ParticipantList` -- a component applicable to both chat and calling settings. Thus, `ChatClientProvider` and `CallingClientProvider` both provide props for this component. Thus, `usePropsFor()` can not determine which provider should be used to satisfy the props. You must explicitly set the target to use:

**Review by prprabhu-ms — APPROVED** [review]

File: packages/storybook/stories/Stateful Client/UsePropsForHook.stories.mdx [hunk 3]
```diff
@@ -8,49 +8,76 @@ import { Canvas, Description, Heading, Props, Source, Title } from '@storybook/a
...
+Since ParticipantList is a shared component between calling and chat, usePropsFor can't determine which side of data you want to collect.
+You need to pass the flavor parameter to usePropsFor() when it happens:
+
+```tsx
+import React from 'react';
+import { ParticipantList, usePropsFor } from '@azure/communication-react';
+
+// This will show 2 different participantLists in parallel
+export const ParicipantLists = () => {
+  const chatPariticipantProps = usePropsFor(ParticipantList, 'chat');
```
Comment (alkwa-msft) [inline packages/storybook/stories/Stateful Client/UsePropsForHook.stories.mdx:70]: chatParticipantProps

**Review by alkwa-msft — APPROVED** [review]

**Review by JamesBurnside — APPROVED** [review]

**Pull request merged**